src/Datadog.Trace/Conventions/OtelOutboundHttpConvention.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.Conventions
+{
+    internal class OtelOutboundHttpConvention : IOutboundHttpConvention
+    {
+        private readonly Tracer _tracer;
+
+        public OtelOutboundHttpConvention(Tracer tracer)
+        {
+            _tracer = tracer;
+        }
+
+        public Scope CreateScope(OutboundHttpArgs args, out HttpTags tags)
+        {
+            var otelTags = new OtelHttpTags();
+            tags = otelTags;
+
+            string operationName = "HTTP " + args.HttpMethod;
+            string serviceName = _tracer.Settings.GetServiceName(_tracer, "http-client");
+            var scope = _tracer.StartActiveWithTags(operationName, tags: tags, serviceName: serviceName, spanId: args.SpanId);
+            scope.Span.Type = SpanTypes.Http;
+
+            var uri = args.RequestUri;
+            otelTags.HttpMethod = args.HttpMethod;
+            otelTags.HttpUrl = string.Concat(uri.Scheme, Uri.SchemeDelimiter, uri.Authority, uri.PathAndQuery, uri.Fragment);
```

Comment:
The OTel. .NET SDK uses `OrginalString`. However, this can cause leakage of auth data.
See: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L115
I feel like this is a little vuln in .NET SDK. Do you agree?

Reply:
Agree we shouldn't use the `OriginalString` approach if this can contain auth data.

Reply:
I asked a question on #otel-specification Slack channel: https://cloud-native.slack.com/archives/C01N7PP1THC/p1614854143044600

Reply:
Good find @pellared definitely do not use `OriginalString`.